optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/bendable/BendableScoreDefinition.java

```diff
@@ -103,8 +105,9 @@ public BendableScore createScoreUninitialized(int initScore, int... scores) {
     }
 
     @Override
-    public BendableScoreInliner buildScoreInliner(boolean constraintMatchEnabled) {
-        return new BendableScoreInliner(constraintMatchEnabled, hardLevelsSize, softLevelsSize);
+    public BendableScoreInliner buildScoreInliner(Map<Constraint, BendableScore> constraintToWeightMap,
+            boolean constraintMatchEnabled) {
```

Comment:
This constraintToWeightMap looks fishy. I 'll need to grok it first.

Reply:
I can get rid of it. See my answer to your score inliner comment.